sql/core/src/main/scala/org/apache/spark/sql/execution/arrow/ArrowConverters.scala

```diff
@@ -226,11 +229,15 @@ private[sql] object ArrowConverters extends Logging {
       schema: StructType,
       timeZoneId: String,
       errorOnDuplicatedFieldNames: Boolean): Array[Byte] = {
-    new ArrowBatchWithSchemaIterator(
+    val batches = new ArrowBatchWithSchemaIterator(
         Iterator.empty, schema, 0L, 0L,
         timeZoneId, errorOnDuplicatedFieldNames, TaskContext.get()) {
       override def hasNext: Boolean = true
-    }.next()
+    }
+    val emptyBatch = batches.next()
```

Comment:
```scala
Utils.tryWithSafeFinally {
  batches.next()
} {
  if ( TaskContext.get() == null) { // If taskContext is not null, it can be successfully closed, right?
    batches.close()
  }
}
```

Reply:
It's right